pkgs/stdenv/generic/make-derivation.nix

```
@@ -163,6 +163,31 @@ let
 
 , env ? { }
 
+, src ? null
+
+, srcWorkDir ? null
+
+, sourceRoot ?
+    if srcWorkDir == null then
+      null
+    else if ! builtins.isPath srcWorkDir then
+      throw "stdenv.mkDerivation: The `srcWorkDir` attribute is expected to be a path, but it's a ${builtins.typeOf srcWorkDir} instead."
+    else if ! builtins.pathExists srcWorkDir then
+      throw "stdenv.mkDerivation: The `srcWorkDir` attribute \"${toString srcWorkDir}\" does not exist."
+    else if lib.filesystem.pathType srcWorkDir != "directory" then
+      throw "stdenv.mkDerivation: The `srcWorkDir` attribute \"${toString srcWorkDir}\" is not a directory"
```

Nit:

```suggestion
      throw "stdenv.mkDerivation: The `srcWorkDir` attribute \"${toString srcWorkDir}\" is not referring to a directory"
```

We use "is a" for the type of attributes, apparently.